clients/fides-js/src/lib/cookie.ts

```diff
@@ -215,26 +211,33 @@ export const saveFidesCookie = (
   // eslint-disable-next-line no-param-reassign
   cookie.fides_meta.updatedAt = updatedAt;
 
-  // Write the cookie to the root domain
-  const rootDomain = window.location.hostname.split(".").slice(-2).join(".");
-
   let encodedCookie: string = JSON.stringify(cookie);
   if (base64Cookie) {
     encodedCookie = base64_encode(encodedCookie);
   }
 
-  setCookie(
-    CONSENT_COOKIE_NAME,
-    encodedCookie,
-    {
-      // An explicit path ensures this is always set to the entire domain.
-      path: "/",
-      // An explicit domain allows subdomains to access the cookie.
-      domain: rootDomain,
-      expires: CONSENT_COOKIE_MAX_AGE_DAYS,
-    },
-    CODEC
-  );
+  const hostname = window.location.hostname.split(".");
+  let topViableDomain = "";
+  for (let i = 1; i < hostname.length + 1; i += 1) {
```

Comment:
```suggestion
  for (let i = 1; i <= hostname.length; i += 1) {
```
I'm sorry this is so nitpicky but I had to